server/src/handlers/auth_handler.rs

```diff
@@ -287,6 +287,8 @@ pub struct AuthQuery {
     pub organization_id: Option<uuid::Uuid>,
     /// URL to redirect to after successful login
     pub redirect_uri: Option<String>,
+    /// URL to redirect to if the user is new
+    pub new_user_redirect_uri: Option<String>,
```

Comment:
I think instead of adding it like this, if it is a new user we should modify the rust code to add a query param to the redirect that indicates a new user was created